### PR TITLE
Blacksmithing 2

### PR DESCRIPTION
**Pull request by MadmanMartian**
Adds percussion-based blacksmithing (Hit it until it looks like what you wanted)

total changes:
 - 2 new macros: ishammer for if your weapon can be used for percussive maintenance, and hasanvil to check if your turf has an anvil on it.
 - some more meltpoint defines
 - A new stack_recipe subtype where it puts the item into a blacksmithing placeholder, where it needs to be heated and hammered into shape
 - A blacksmithing_placeholder that will hold the result, the whacks, etc.

## Sprites requested
 - [x] An improvement over my current sprites (The hammer and the chain)
 - [x] A forge, preferably something that looks like it could be fuelled with plasma ore/sheets
 - [x] Some chainmail armor sprites

:cl:
 * rscadd: Adds blacksmithing

### CODE REVIEW

**Comment by MadmanMartian** [comment]
tfw the elder god of machinery approves of your PR

**Comment by RealestEstate** [comment]
chainmail
I pinged you with this updated version in ivory tower discord but I don't know if you saw it since I noticed you were in Do Not Disturb mode. I'm pretty sure if you change the extension from .png to .dmi this should perfectly switch over.

**Comment by SonixApache** [comment]
gonfligds :DDD

File: __DEFINES/reagents.dm [hunk 1]
```diff
@@ -396,6 +396,7 @@
 #define TEMPERATURE_FLAME 700
 #define TEMPERATURE_WELDER 3480
 #define TEMPERATURE_PLASMA 4500
+#define TMEPERATURE_ETHANOL (T0C+1560)
```
Comment (jknpj) [inline __DEFINES/reagents.dm]: typo

File: __DEFINES/_macros.dm [hunk 2]
```diff
@@ -132,6 +132,8 @@
 
 #define iswelder(A) istype(A, /obj/item/weapon/weldingtool)
 
+#define ishammer(A) is_type_in_list(A, list(/obj/item/weapon/hammer, /obj/item/weapon/storage/toolbox))
```
Comment (jknpj) [inline __DEFINES/_macros.dm:135]: how about using the blunt flag + weight?

Comment (jknpj) [reply __DEFINES/_macros.dm:135]: disregard this, i forgot we have no blunt sharpness flag yet, maybe later on.

File: __DEFINES/reagents.dm (hunk 1 above)
Comment (MadmanMartian) [reply __DEFINES/reagents.dm]: tmep

**Comment by MadmanMartian** [comment]
work continues at the code-forge

File: code/WorkInProgress/MadmanMartian/blacksmithing/blacksmithing.dm [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+/obj/item/smithing_placeholder/afterattack(atom/A, mob/user, proximity_flag, click_parameters)
+	if(!proximity_flag)
+		return
+	if(isobj(A))
+		var/obj/O = A
+		if(O.is_hot())
+			heat(O.is_hot(), O, user)
+		if(O.is_open_container() && O.reagents.has_reagent(WATER, 60))
+			quench(O, user)
+	if(istype(A, /mob/living/simple_animal/hostile/asteroid/magmaw)) //Until we have flameslimes, lavalizards, crimson pyromancers, or flaming skeletons, this will be hardcoded
```
Comment (DamianX) [inline code/WorkInProgress/MadmanMartian/blacksmithing/blacksmithing.dm:51]: `/mob/living/carbon/monkey/skellington/plasma` :^)

File: code/WorkInProgress/MadmanMartian/blacksmithing/forge.dm [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+/obj/structure/forge/update_icon()
+	if(status)
+		icon_state = "furnace_on"
+	else
+		icon_state = "furnace_off"
+
+/obj/structure/forge/Destroy()
+	processing_objects.Remove(src)
+	heating.forceMove(get_turf(src))
```
Comment (DamianX) [inline code/WorkInProgress/MadmanMartian/blacksmithing/forge.dm:27]: Do you expect the forge to only be destroyed when an item is inside it?

File: code/WorkInProgress/MadmanMartian/blacksmithing/forge.dm [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+	heating = null
+	for(var/obj/I in contents)
```
Comment (DamianX) [inline code/WorkInProgress/MadmanMartian/blacksmithing/forge.dm:29]: `/atom/movable/Destroy()` does this

File: code/WorkInProgress/MadmanMartian/blacksmithing/forge.dm [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+		var/obj/item/stack/sheet/mineral/plasma/P = I
+		if(P.use(1))
+			to_chat(user, "<span class = 'notice'>You toss a sheet of \the [I] into \the [src].</span>")
+			if(current_temp < TEMPERATURE_PLASMA)
+				current_temp = TEMPERATURE_PLASMA
+			fuel_time+= 60 SECONDS
+			return
+	else if(istype(I, /obj/item/weapon/ore/plasma))
+		to_chat(user, "<span class = 'notice'>You toss \the [I] into \the [src].</span>")
+		user.drop_item(I)
```
Comment (DamianX) [inline code/WorkInProgress/MadmanMartian/blacksmithing/forge.dm:44]: should probably check for return value

File: code/WorkInProgress/MadmanMartian/blacksmithing/misc_components.dm [hunk 7]
```diff
@@ -0,0 +1,37 @@
+/obj/item/item_head
+	icon = 'icons/obj/misc_components.dmi'
+	var/obj/item/result
+
+/obj/item/item_head/attackby(obj/item/I, mob/user)
+	if(istype(I, /obj/item/item_handle))
+		to_chat(user, "<span class = 'notice'>You begin to attach \the [I] to \the [src].</span>")
+		if(do_after(user, src, 4 SECONDS))
+			user.drop_item(I)
```
Comment (DamianX) [inline code/WorkInProgress/MadmanMartian/blacksmithing/misc_components.dm:9]: should probably check return value

File: code/WorkInProgress/MadmanMartian/blacksmithing/misc_components.dm [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+			user.drop_item(src)
```
Comment (DamianX) [inline code/WorkInProgress/MadmanMartian/blacksmithing/misc_components.dm:10]: should probably check return value

**Comment by io-june** [comment]
anvilposting will rise again

**Pull request merged**